manager2/src/app/tps/tps.component.ts

```diff
@@ -174,6 +180,25 @@ export class TpsComponent implements OnInit {
         this.listEvents();
     }
 
+    extend_reservation(new_expire: Date) {
```

Comment:
You're passing 'new_expire' here (and on the html side), but you are using this.new_expire instead.

I'm not sure if there is a point in passing it?